PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp

```diff
@@ -14,8 +14,9 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-#include <shlobj.h>
+#include <shlwapi.h>
 #include <uxtheme.h>
+#include "Common.h"

```

Comment:
`Common.h` is not needed.
Please remove it.

Reply:
It is needed because I used `PathRemoveFileSpec()` overload for `generic_string` declared in `"Common.h"`.

Reply:
The signature of common.h's version is `PathRemoveFileSpec(generic_string&)`
shlwapi.h's version one is `PathRemoveFileSpec(TCHAR*)`
The `PathRemoveFileSpec` you use here passed a generic_string object as argument so there's no ambiguity.
"Common.h" is already included in "localization.h" which is included in "FindReplaceDlg.cpp". 
So even you remove the line `#include "Common.h"`, compiler can link to the right function.
For the sake of maintenance facility, in both *.h & *.cpp, we try to include as minimun header as passible.

Reply:
I think this is micro-management.
Each header should have either [pragma once](https://en.wikipedia.org/wiki/Pragma_once) or [include guard](https://en.wikipedia.org/wiki/Include_guard) (`Common.h` has `#pragma once`).
So it is the task of the compiler (not the programmer) to handle the fact that a file is included multiple times.

Each file should include all headers it requires to make it self-sufficient. It is bad to rely on includes in other headers. What if `Common.h` will be removed from `localization.h`? Then you'll have to fix every place that was relying on the fact that it included `Common.h`. This makes the code really fragile.

On the other hand, if you're _really_ concerned with the build time, I'd suggest you to create a dedicated GitHub issue.
Be sure, I have some ideas for that too.

Reply:
It's not even a "management". I try to make all contributed code into master conform my standard/style so I could maintain in the future.

> Each file should include all headers it requires to make it self-sufficient. It is bad to rely on includes in other headers. What if Common.h will be removed from localization.h? Then you'll have to fix every place that was relying on the fact that it included Common.h. This makes the code really fragile.

I got your point but I don't consider it's "fragile". The situation you described comes only on compiling time but not on the runtime - the dev adds the header needed and the compiling problem solved.

> On the other hand, if you're really concerned with the build time,  I'd suggest you to create a dedicated GitHub issue.
Be sure, I have some ideas for that too.

The goal of including as less headers as possible is for the maintenance, not for reducing the build time. I care about runtime whereas the build time it doesn't matter to me. But feel free to create issues if you have some suggestions by tagging me.